shared/src/commonMain/kotlin/co/nimblehq/avishek/phong/kmmic/presentation/module/HomeViewModel.kt

```diff
@@ -0,0 +1,51 @@
+package co.nimblehq.avishek.phong.kmmic.presentation.module
+
+import co.nimblehq.avishek.phong.kmmic.domain.usecase.GetSurveysUseCase
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.catch
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
+import kotlinx.coroutines.flow.onStart
+import kotlinx.coroutines.flow.update
+
+data class HomeViewState(
+    val isLoading: Boolean
+) {
+    constructor() : this (true)
+}
+
+class HomeViewModel(
+    private val getSurveysUseCase: GetSurveysUseCase
+) : BaseViewModel() {
+
+    private val _viewState: MutableStateFlow<HomeViewState> =
+        MutableStateFlow(HomeViewState())
+
+    val viewState: StateFlow<HomeViewState> = _viewState
+
+    private var currentPage = 1
+
+    fun fetchData() {
+        getSurveysUseCase(currentPage, 10, isForceLatestData = false)
```

Comment:
Shall we make a const for `pageSize`?
```suggestion
        getSurveysUseCase(pageNumber = currentPage, pageSize = DEFAULT_PAGE_SIZE, isForceLatestData = false)
```